phoenix-core-client/src/main/java/org/apache/phoenix/compile/UpsertCompiler.java

```diff
@@ -636,7 +636,7 @@ public MutationPlan compile(UpsertStatement upsert) throws SQLException {
                 // as max TS, so that the query can safely restarted and still work of a snapshot
                 // (so it won't see its own data in case of concurrent splits)
                 // see PHOENIX-4849
-                long serverTime = selectResolver.getTables().get(0).getCurrentTime();
+                long serverTime = selectResolver.getTables().get(0).getTimeStamp();
```

Comment:
What is the difference between getCurrentTime() and getTimeStamp() ?

Reply:
@tkhurana Currently both of those methods return the same value i.e. `upperBoundTimestamp` which is either -1 or whatever is provided to the TableRef constructor. Current code:
```
// if UPDATE_CACHE_FREQUENCY is set, always let the server set timestamps
this.upperBoundTimeStamp = table.getUpdateCacheFrequency()!=0 ? QueryConstants.UNSET_TIMESTAMP : upperBoundTimeStamp;
this.currentTime = this.upperBoundTimeStamp;

public long getTimeStamp() {
        return this.upperBoundTimeStamp;
}
public long getCurrentTime() {
        return this.currentTime;
 }
```
Once we set UPDATE_CACHE_FREQ to NEVER, currentTime was being set to -1 and features using currentTime were breaking. This change keeps currentTime to whatever is provided to the TableRef constructor so that features using currentTime can continue to use it. There is effectively no change in UpsertCompiler - it will still be using the same value i.e. TableRef.upperBoundTimeStamp.
```
this.currentTime = upperBoundTimeStamp;
```